File: twtcinema/src/apiService/episode.js
import instance from './instance';

export const uploadEpisode = async (formData) => {
    return instance.post('/episode/upload', formData, {
        headers: {
            'Content-Type': 'multipart/form-data',
        },
    });
};

export const getEpisodes = async (movieId) => {
    return instance.get('/episode/get-all/' + movieId);
};

export const getEpisodeData = async (movieId, epNum) => {
    return instance.get(`/episode/get-ep-data/${movieId}/${epNum}`);
};

export const editEpisode = async (ep_id, formData) => {
    return instance.post('/episode/edit-ep/' + ep_id, formData);
};

File: flask-server/src/routes/episode.py
from controllers.episode import upload_episode, get_all_episodes, get_ep_data

from flask import Blueprint, Flask


def register_video_routes(app: Flask):
    # Create a Blueprint for API routes
    api_blueprint = Blueprint("episode", __name__, url_prefix="/api/episode")

    # Define API routes
    api_blueprint.add_url_rule("/upload", view_func=upload_episode, methods=["POST"])
    api_blueprint.add_url_rule(
        "/get-all/<string:movie_id>", view_func=get_all_episodes, methods=["GET"]
    )
    api_blueprint.add_url_rule(
        "/get-ep-data/<string:movie_id>/<string:ep_num>",
        view_func=get_ep_data,
        methods=["GET"],
    )

    # Register Blueprint
    app.register_blueprint(api_blueprint)
